components/brave_wallet_ui/common/hooks/swap.ts

```diff
@@ -412,10 +425,12 @@ export default function useSwap ({ fromAsset: fromAssetProp, toAsset: toAssetPro
         // order to ensure a swap with minimum slippage.
         const { estimation: gasEstimates } = await ethTxManagerProxy.getGasEstimation1559()
 
-        const isEIP1559 = hasEIP1559Support(selectedAccount, selectedNetwork) && gasEstimates
+        const isEIP1559 = selectedNetwork && gasEstimates
+          ? hasEIP1559Support(selectedAccount, selectedNetwork)
+          : false
         let maxPriorityFeePerGas
         let maxFeePerGas
-        if (isEIP1559 && gasEstimates.fastMaxPriorityFeePerGas === gasEstimates.avgMaxPriorityFeePerGas) {
+        if (isEIP1559 && gasEstimates && gasEstimates.fastMaxPriorityFeePerGas === gasEstimates.avgMaxPriorityFeePerGas) {
```

Comment:
```suggestion
        if (isEIP1559 && gasEstimates?.fastMaxPriorityFeePerGas === gasEstimates.avgMaxPriorityFeePerGas) {
```

Reply:
The reason why I did this way was so that the `gasEstimates &&` indicates tsc that `gasEstimates` is truthy inside the if-block.

One other issue with your version is there's a chance that `undefined === undefined` condition gets satisfied.